src/export/ExportWavPack.cpp

```diff
@@ -0,0 +1,502 @@
+      config.flags |= CONFIG_FAST_FLAG;
+   } else if (quality == 1) {
+      config.flags |= CONFIG_HIGH_FLAG;
+   } else {
+      config.flags |= CONFIG_VERY_HIGH_FLAG;
+   }
+
+   if (hybridMode) {
+      config.flags |= CONFIG_HYBRID_FLAG;
+      config.flags |= CONFIG_BITRATE_KBPS;
+      config.bitrate = bitRate;
+
+      if (createCorrectionFile) {
+         config.flags |= CONFIG_CREATE_WVC;
+
+         outWvcFile.file = std::make_unique< wxFile >();
+         if (!outWvcFile.file->Create(fName.GetFullPath().Append("c"), true)) {
+            AudacityMessageBox( XO("Unable to create target file for writing") );
+            return ProgressResult::Failed;
+         }
+      }
+   }
+
+   WavpackContext *wpc = WavpackOpenFileOutput(WriteBlock, &outWvFile, createCorrectionFile ? &outWvcFile : nullptr);
+   auto closeWavPackContext = finally([wpc]() { WavpackCloseFile(wpc); });
+
+   if (!WavpackSetConfiguration64(wpc, &config, -1, nullptr) || !WavpackPackInit(wpc)) {
+      ShowExportErrorDialog( WavpackGetErrorMessage(wpc) );
+      return ProgressResult::Failed;
+   }
+
+   // Samples to write per run
+   constexpr size_t SAMPLES_PER_RUN = 8192u;
+
+   const size_t bufferSize = SAMPLES_PER_RUN * numChannels;
+   ArrayOf<int32_t> wavpackBuffer{ bufferSize };
+   auto updateResult = ProgressResult::Success;
+   {
+      auto mixer = CreateMixer(tracks, selectionOnly,
+         t0, t1,
+         numChannels, SAMPLES_PER_RUN, true,
+         rate, format, mixerSpec);
+
+      InitProgress( pDialog, fName,
+         selectionOnly
+            ? XO("Exporting selected audio as WavPack")
+            : XO("Exporting the audio as WavPack") );
+      auto &progress = *pDialog;
+
+      while (updateResult == ProgressResult::Success) {
+         auto samplesThisRun = mixer->Process(SAMPLES_PER_RUN);
+
+         if (samplesThisRun == 0)
+            break;
+         
+         if (format == int16Sample) {
+            const char *mixed = mixer->GetBuffer();
+            for (decltype(samplesThisRun) j = 0; j < samplesThisRun; j++) {
+               for (size_t i = 0; i < numChannels; i++) {
+                  int32_t value = *mixed++ & 0xff;
+                  value += *mixed++ << 8;
+                  wavpackBuffer[j*numChannels + i] = value;
+               }
+            }
+         } else if (format == int24Sample) {
+            const int *mixed = reinterpret_cast<const int*>(mixer->GetBuffer());
+            for (decltype(samplesThisRun) j = 0; j < samplesThisRun; j++) {
+               for (size_t i = 0; i < numChannels; i++) {
+                  wavpackBuffer[j*numChannels + i] = *mixed++;
+               }
+            }
+         } else {
+            const float *mixed = reinterpret_cast<const float*>(mixer->GetBuffer());
+            for (decltype(samplesThisRun) j = 0; j < samplesThisRun; j++) {
+               for (size_t i = 0; i < numChannels; i++) {
+                  int64_t intValue = (*mixed++) * (1u << 31);
```

Comment:
In the `ExportWavPack.cpp` file, I think there were some issues with float 32 bit export. Earlier, I just passed 32 bit int values for float sample export. So, I think while importing it was getting imported as 32 bit int (before adding special cases for float sample import as you suggested). Though it was ok while importing in audacity ( wtih the previous code before applying your suggestions for import), I tried to play the 32 bit exported wavpack file using the VLC media player and it sounded like corrupted. Now, when I converted the `mixed` data into `float` and then into `int32_t` while exporting as 32 bit float, both VLC media and audacity (After applying your suggestions) opened and played the 32 bit float files correctly.

Reply:
I think you should set `config->float_norm_exp: = 127;` to enable FP mode in Wavpack. In any case - you should consult `WavpackGetMode` to check, if you need to convert the values.

`1u << 31` - `std::numeric_limits<int32_t>::max()` please.

Reply:
(Value `127` is an exponent  bias for IEEE754 32-bit float values. We have 8 bits for exponents, so the bias is `2^7 - 1`)